syspage.c

```diff
@@ -494,13 +501,16 @@ int syspage_validateKernel(addr_t *addr)
 
 int syspage_addmap(const char *name, addr_t start, addr_t end, const char *attr)
 {
-	int i;
+	int i, res;
 	size_t size;
 	map_entry_t *entry;
 	syspage_map_t *map;
 	plo_map_t *ploMap;
 	u32 len, mapID = syspage_common.mapsCnt;
 
+	if (mapID >= MAX_MAPS_NB)
+		return -EFAULT;
+
```

Comment:
Such comparison should be implemented only in hal. MAX_MAPS_NB macro is hardware dependent.

Reply:
Removed.